### ISSUE TITLE
exceptions.AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'

### ISSUE THREAD

**Issue by kevinoid** [nabla-c0d3/sslyze#112]
Running `sslyze --regular --starttls=xmpp kevinlocke.name:5222` with [SSLyze v0.12](https://github.com/nabla-c0d3/sslyze/releases/download/release-0.12/sslyze-0_12-linux64.zip) results in output which has this error message scattered throughout:

```



 AVAILABLE PLUGINS
 -----------------

  PluginCompression
  PluginHeartbleed
  PluginOpenSSLCipherSuites
  PluginCertInfo
  PluginSessionRenegotiation
  PluginChromeSha1Deprecation
  PluginHSTS
  PluginSessionResumption



 CHECKING HOST(S) AVAILABILITY
 -----------------------------

   kevinlocke.name:5222                => 2605:6400:2:fed5:22:0:f2f6:c0f7:5222



 SCAN RESULTS FOR KEVINLOCKE.NAME:5222 - 2605:6400:2:FED5:22:0:F2F6:C0F7:5222
 ----------------------------------------------------------------------------

Unhandled exception when processing --reneg: 
exceptions.AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'

Unhandled exception when processing --compression: 
exceptions.AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'

  * OpenSSL Heartbleed:
      OK - Not vulnerable to Heartbleed  

Unhandled exception when processing --certinfo: 
exceptions.AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'

  * Session Resumption:
      With Session IDs:                  ERROR (0 successful, 0 failed, 5 errors, 5 total attempts).
        ERROR #1: AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
        ERROR #2: AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
        ERROR #3: AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
        ERROR #4: AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
        ERROR #5: AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
      With TLS Session Tickets:          ERROR: AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'

  * SSLV2 Cipher Suites:
      Server rejected all cipher suites.

  * TLSV1_2 Cipher Suites:
      Undefined - An unexpected error happened: 
                 ECDHE-RSA-AES256-SHA384             AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 ECDHE-RSA-AES256-SHA                AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 ECDHE-RSA-AES256-GCM-SHA384         AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 ECDHE-RSA-AES128-SHA256             AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 ECDHE-RSA-AES128-SHA                AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 ECDHE-RSA-AES128-GCM-SHA256         AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 CAMELLIA256-SHA                     AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 CAMELLIA128-SHA                     AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 AES256-SHA256                       AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 AES256-SHA                          AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 AES256-GCM-SHA384                   AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 AES128-SHA256                       AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 AES128-SHA                          AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 AES128-GCM-SHA256                   AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'

  * TLSV1_1 Cipher Suites:
      Undefined - An unexpected error happened: 
                 ECDHE-RSA-AES256-SHA                AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 ECDHE-RSA-AES128-SHA                AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 CAMELLIA256-SHA                     AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 CAMELLIA128-SHA                     AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 AES256-SHA                          AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 AES128-SHA                          AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'

  * TLSV1 Cipher Suites:
      Undefined - An unexpected error happened: 
                 ECDHE-RSA-AES256-SHA                AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 ECDHE-RSA-AES128-SHA                AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 CAMELLIA256-SHA                     AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 CAMELLIA128-SHA                     AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 AES256-SHA                          AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'
                 AES128-SHA                          AttributeError - 'XMPPConnection' object has no attribute 'get_client_ca_list'

  * SSLV3 Cipher Suites:
      Server rejected all cipher suites.



 SCAN COMPLETED IN 29.83 S
 -------------------------
```

After a little digging and testing, I realized that this is solved by applying the fix from nabla-c0d3/nassl#35.  I'm not sure if the issue is XMPP in general, or that Prosody is requesting client certificates and exercising an uncommon codepath.  In either case, it would be great to see a new release which fixes this issue so new users can avoid troubleshooting the issue.

Thanks,
Kevin

**Comment by nabla-c0d3** [comment]
Yes - the fix will be released in 0.13. It happens when the server requests a client certificate. Thanks for looking into it.

**Comment by kevinoid** [comment]
Good to know, thanks!

**Comment by nabla-c0d3** [comment]
Fixed in 5ed70d8.

**Comment by kevinoid** [comment]
Thanks @nabla-c0d3!
